src/tools/run-make-support/src/command.rs

```diff
@@ -151,6 +151,53 @@ impl Command {
         self
     }
 
+    /// Set an auxiliary stream passed to the process, besides the stdio streams.
+    ///
+    /// # Safety
+    ///
+    /// Use with caution! Ideally, only set one aux fd; if there are multiple, their old `fd` may
+    /// overlap with another's `new_fd`, and may break. The caller must make sure this is not the
+    /// case.
+    #[cfg(unix)]
+    pub unsafe fn set_aux_fd<F: Into<std::os::fd::OwnedFd>>(
+        &mut self,
+        new_fd: std::os::fd::RawFd,
+        fd: F,
+    ) -> &mut Self {
+        // NOTE: If more than 1 auxiliary file descriptor is needed, this function should be
+        // rewritten.
+
+        use std::os::fd::AsRawFd;
+        use std::os::unix::process::CommandExt;
+
+        let cvt = |x| if x == -1 { Err(std::io::Error::last_os_error()) } else { Ok(()) };
+
+        // Ensure fd stays open until the fork.
+        let fd = mem::ManuallyDrop::new(fd.into());
+        let fd = fd.as_raw_fd();
+
+        if fd == new_fd {
+            // If the new file descriptor is already the same as fd, just turn off `FD_CLOEXEC`.
+
+            // SAFETY(io-safety): `fd` is already owned.
+            cvt(unsafe { libc::fcntl(fd, libc::F_SETFD, 0) }).expect("disabling CLOEXEC failed");
```

Comment:
nit: it's more future compatible to first `F_GETFD`, unset the bit, and throw that in there instead of 0

Reply:
Changed this to first retrieve the fd flags, then clear the `FD_CLOEXEC` bit, instead of YOLOing a `0` when `F_SETFD`.